device/api/umd/device/types/arch.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2024 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <ostream>
+
+namespace tt::umd {
+
+static inline std::string to_lower(const std::string &str) {
+    std::string res = str;
+    std::transform(res.begin(), res.end(), res.begin(), ::tolower);
+    return res;
+}
+
+/**
+ * Enums for different architectures.
+ */
+enum class Arch {
+    GRAYSKULL = 1,
+    WORMHOLE_B0 = 2,
+    BLACKHOLE = 3,
+    Invalid = 0xFF,
+};
+
+static inline tt::umd::Arch arch_from_str(const std::string &arch_str) {
```

Comment:
After https://github.com/tenstorrent/tt-umd/pull/345, consolidate with this function